test/CC/CFLAGS-live.py

```diff
@@ -23,14 +23,21 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+"""
+Test behavior of CFLAGS.
+
+This is a live test, uses the detected C compiler.
+"""
+
 import sys
+
 import TestSCons
 
 test = TestSCons.TestSCons()
 
 # Make sure CFLAGS is not passed to CXX by just expanding CXXCOM
-test.write('SConstruct', """
-DefaultEnvironment(tools=[])
+test.write('SConstruct', """\
+_ = DefaultEnvironment(tools=[])
```

Comment:
Why do this? lint complaining?

Reply:
More for editors... it returns an env, so there's a grumble if you ignore that it has a return. Won't trigger in cases like these, where it's code embedded in a string to `test.write`, but I'm trying to keep up the habit.

Reply:
I'd rather not have any code with x=DefaultEnvironment().. users already do this, and I don't think it's good practice.